PowerEditor/src/NppNotification.cpp

```diff
@@ -946,6 +946,17 @@ BOOL Notepad_plus::notify(SCNotification *notification)
 			if (not notifyView)
 				return FALSE;
 
+			// Check if a restore position is needed. 
+			// Restoring a position must done after SCN_PAINTED notification so that it works in every circumstances (including wrapped large file)
+			if (_mainEditView.isPositionRestoreNeeded())
+			{
+				_mainEditView.restoreCurrentPosAfterPainted();

```

Comment:
Here we do restoreCurrentPosAfterPainted() then we do restoreCurrentPos() (line #964) - it's quite confused for me.
Could you enlight me about this point please?

Reply:
> First step: set selection, set anchor, set xoffset...
> Second step: once Scintilla has send the notification SCN_PAINTED, we can scroll several lines to set the first visible line to the correct wrapped line.

So if I do understand, first step is `restoreCurrentPos()` (line number 964) to enable `positionRestoreNeeded`, then the SCN_PAINTED message will be sent again.
`restoreCurrentPosAfterPainted()` is the 2nd step ?

Reply:
Oui, vous avez totalement raison. Juste une précision, scintilla peut envoyer plusieurs noticiation SCN_PAINTED avant que le buffer soit prés à êter afficher. cette seconde étape est donc itérée plusieurs fois dans un maximum de 7 itérations. 7 est un chiffre arbitraire que j'ai choisi. 2 est un minimum (trouvé avec des tests). Je ne connais pas de maximum.

Pour information, dans mon PR précédent, les tests que tu as fait étaient KO car cette seconde étape n'était pas itérable.

Reply:
The name of function `restoreCurrentPosAfterPainted` is not explicit to me and bring some confusion.
Do we have another better naming for this function?